src/components/Q32Rectangles/index.tsx

```diff
@@ -0,0 +1,40 @@
+import { useCallback } from 'react';
+
+interface IQ32RectanglesProps {
+    x: number;
+    width: number;
+    height: number;
+    color: string;
+}
+
+const Q32Rectangles: React.FC<IQ32RectanglesProps> = ({
+    x,
+    width,
+    height,
+    color,
+}) => {
+    const rectangleArray = [] as number[];
+
+    const buildRectanglesArray = useCallback(() => {
+        for (let i = 1; i <= x; i += 1) {
+            rectangleArray.push(i);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    buildRectanglesArray();
```

Comment:
Isso deveria ser feito usando o `useEffect`, porque é algo que acontece 1x no início do componente. O `useCallback` é pra ser executado sempre que você interage com algum elemento, por exemplo um clique em botão